From: yangyanwei <yangyanwei@kylinos.cn>
Date: Thu, 20 Aug 2026 09:54:33 +0800
Subject: =?utf-8?b?Zml4OiDkv67lpI3nnIvlm77miZPlvIDmlofku7blr7nor53moYblpZQ=?=
 =?utf-8?b?5rqD6Zeu6aKY?=
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

note:
ListView::bindModel 在 libpeony-qt/controls/directory-view/view/list-view/list-view.cpp 调用 m_model->setCheckboxColumnEnabled(true)，此时模型刚绑定到视图，可能还没有 root item（m_root_item 为 nullptr）

Bug: #640839
From: kylin
Severity: Moderate
Changelog-Bug: #640839 【ukui4.24】【需求72731】【文件管理器】列表模式下，文件名称左侧全选框显示不全
Change-Id: I99810f64f6533050e229c54173bdebc289224bd1
---
 libpeony-qt/model/file-item-model.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libpeony-qt/model/file-item-model.cpp b/libpeony-qt/model/file-item-model.cpp
index 53cb463..562a011 100644
--- a/libpeony-qt/model/file-item-model.cpp
+++ b/libpeony-qt/model/file-item-model.cpp
@@ -164,6 +164,10 @@ void FileItemModel::setRootItem(FileItem *item)
 
 QModelIndex FileItemModel::index(int row, int column, const QModelIndex &parent) const
 {
+    if (!m_root_item) {
+        return QModelIndex();
+    }
+
     //root children
     if (!parent.isValid()) {
         if (row < 0 || row > m_root_item->m_children->count()-1)
